### PR TITLE
Update samples and test sites to new formats

### LINKED ISSUE #4428: Update sample projects
Going forward all sample projects should have these changes:
In your Startup.cs you must include
```
new WebHostBuilder().UseContentRoot(Directory.GetCurrentDirectory()).UseDefaultHostConfiguration(args)
```
in your project.json you need
```
"content": [
    "wwwroot",
    "Views",
    "appsettings.json",
    "web.config"
],
```
Samples:
- [x] ActionConstraintSample.Web
- [x] CustomRouteSample.Web	
- [x] EmbeddedViewSample.Web
- [x] FormatFilterSample.Web
- [x] InlineConstraintSample.Web
- [x] JsonPatchSample.Web
- [x] LocalizationSample.Web
- [x] MvcSandbox
- [x] MvcSubAreaSample.Web
- [x] TagHelperSample.Web
- [x] UrlHelperSample.Web

Test websites:
- [x] ApiExplorerWebSite
- [x] ApplicationModelWebSite
- [x] BasicWebSite
- [x] ControllersFromServicesClassLibrary
- [x] ControllersFromServicesWebSite
- [x] CorsWebSite
- [x] ErrorPageMiddlewareWebSite
- [x] FilesWebSite
- [x] FiltersWebSite
- [x] FormatterWebSite
- [x] HtmlGenerationWebSite
- [x] Microsoft.AspNetCore.Mvc.TestConfiguration
- [x] RazorPageExecutionInstrumentationWebSite
- [x] RazorWebSite
- [x] RoutingWebSite
- [x] SimpleWebSite
- [x] TagHelpersWebSite
- [x] UserClassLibrary
- [x] VersioningWebSite
- [x] WebApiCompatShimWebSite
- [x] XmlFormattersWebSite

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #4428.

CC @NTaylorMullen

### CODE REVIEW

File: samples/InlineConstraintSample.Web/Startup.cs [hunk 1]
```diff
@@ -14,11 +15,11 @@ public class Startup
     {
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddRouting( routeOptions =>
-            {
-                routeOptions.ConstraintMap.Add("IsbnDigitScheme10", typeof(IsbnDigitScheme10Constraint));
-                routeOptions.ConstraintMap.Add("IsbnDigitScheme13", typeof(IsbnDigitScheme13Constraint));
-            });
+            services.AddRouting(routeOptions =>
```
Comment (NTaylorMullen) [inline samples/InlineConstraintSample.Web/Startup.cs]: Either this line or the one following has odd spacing

Comment (ryanbrandenburg) [inline samples/InlineConstraintSample.Web/Startup.cs]: Bah, VS stuck it's nose where I didn't ask it to. Will fix one way or another.

**Comment by NTaylorMullen** [comment]
Hmm since your touching these give these other expectations a shot and see if they work:

- Target `netcoreapp1.0` instead of `netstandardapp1.5`
- Add imports for the dotnet-razor-tooling tool entries. I think net45 will do.

**Comment by dougbu** [comment]
Along the same lines as @NTaylorMullen's comment, could also
- Add `.UseIISIntegration()` to the `Main` chains
- Remove leftover `.UseIISPlatformHandlerUrl()` calls in the `Main` chains
- Add `"Microsoft.AspNetCore.Server.IISIntegration": "1.0.0-*"` to the dependencies
- Remove leftover `"Microsoft.AspNetCore.IISPlatformHandler": "1.0.0-*"` dependencies

If you do _not_ do this, please file an issue for it.

**Comment by dougbu** [comment]
@ryanbrandenburg ignore [my previous comment](https://github.com/aspnet/Mvc/pull/4429#issuecomment-207563649). Looks as if @JunTaoLuo is doing that work in #4433. (One of you will have a rather large number of merge conflicts I imagine :smiling_imp:)

**Comment by JunTaoLuo** [comment]
@ryanbrandenburg I have merged my changes for the mvc samples. I just realized that there is also a set of test websites that require the same updates. Since I don't want to keep giving you merge conflicts, can you update those as part of this change? Let me know if you have any questions on what changes to make. Thanks!

**Comment by javiercn** [comment]
@JunTaoLuo We don't run tests on IIS but we want the websites to be launchable on IIS, hence the web.config

**Comment by JunTaoLuo** [comment]
@javiercn even the test websites? If so we should add or move the `web.config` to the application root.

The consistency here isn't so great either. Why do only 3 of them have `web.config`s? Some of them makes sense not having one like `UserClassLibrary` but I think a lot of test sites like `RoutingWebSite` would want to be launchable via IIS. There is also placeholder `HelloWorld.htm`s which should be cleaned up. And the `command` nodes of `project.json` should be removed from test sites too.

**Comment by ryanbrandenburg** [comment]
@JunTaoLuo you're probably right about those things, but I'd rather not muddle them into an already large PR since they're not very related (being about cleanup vs updating to a new ways of doing things).

**Comment by ryanbrandenburg** [comment]
Oh I see, those things are related to your changes that you're wanting me to include in here, not to updating the sites to new formats.

I think overall it would be faster and more accurate for you to handle those changes since you already know the nuances, and I don't mind merge conflicts. If you're to swamped with build blockers and other rc2 issues however I can take it anyway.

**Comment by ryanbrandenburg** [comment]
:up::date:

File: samples/MvcSandbox/project.json [hunk 2]
```diff
@@ -17,14 +17,24 @@
...
     "Views",
-    "web.config"
+    "web.config",
+    "wwwroot"
   ],
+  "tools": {
+    "dotnet-razor-tooling": {
+      "version": "1.0.0-*",
+      "imports": [
+        "net45"
```
Comment (NTaylorMullen) [inline samples/MvcSandbox/project.json]: Does this bit actually work? Can verify by looking at the lock file and verifying that it says that dotnet-razor-tooling is a lock file and not `{}`

Comment (ryanbrandenburg) [inline samples/MvcSandbox/project.json]: Yeah, seems to work, it's not empty in the lock file.

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by JunTaoLuo** [comment]
@ryanbrandenburg I think you'll need a rebase after I merged my changes...

**Comment by ryanbrandenburg** [comment]
closing because I have a seperate release PR in #4503

**Pull request closed without merging**